rotkehlchen/db/filtering.py

```diff
@@ -84,23 +84,33 @@ def prepare(self) -> Tuple[List[str], List[Any]]:
 
 
 @dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
-class DBETHTransactionAddressFilter(DBFilter):
+class DBETHTransactionJoinsFilter(DBFilter):
     """Find transactions involving any of the addresses. Including in an internal.
 
     This uses the mappings we create in the DB at transaction addition to signify
     relevant addresses for a transaction.
     """
-    addresses: List[ChecksumEthAddress]
+    addresses: Optional[List[ChecksumEthAddress]]
```

Comment:
Also update docstring then explaining the different uses of this filter and what do the combinations mean. What is join_events etc.

Reply:
Yep, updated